src/keyring/KeyringController.ts

```diff
@@ -421,6 +433,25 @@ export class KeyringController extends BaseController<
   ) {
     try {
       const address = normalizeAddress(messageParams.from);
+      const qrKeyring = await this.getOrAddQRKeyring();
+      const qrAccounts = await qrKeyring.getAccounts();
+      if (
+        qrAccounts.findIndex(
+          (qrAddress: string) =>
+            qrAddress.toLowerCase() === address.toLowerCase(),
+        ) !== -1
+      ) {
+        const messageParamsClone = { ...messageParams };
+        if (
+          version !== SignTypedDataVersion.V1 &&
+          typeof messageParamsClone.data === 'string'
+        ) {
+          messageParamsClone.data = JSON.parse(messageParamsClone.data);
+        }
+        return privates
+          .get(this)
+          .keyring.signTypedMessage(messageParamsClone, { version });
+      }
```

Comment:
Mind adding an extra blank line after this line? I think it'll make the code look less smushed.